File: cinemax-theater-be/src/main/java/fu/se/cinemaxtheaterbe/features/room/services/RoomService.java
package fu.se.cinemaxtheaterbe.features.room.services;

import fu.se.cinemaxtheaterbe.features.room.dtos.RoomDetailResponse;
import fu.se.cinemaxtheaterbe.features.room.dtos.RoomRequest;
import fu.se.cinemaxtheaterbe.features.room.dtos.RoomResponse;

import java.util.List;

public interface RoomService {

    List<RoomResponse> getAllRooms();

    RoomResponse getRoomById(Long id);

    RoomDetailResponse getRoomDetail(Long id);

    RoomResponse createRoom(RoomRequest request);

    RoomResponse updateRoom(Long id, RoomRequest request);

    void deleteRoom(Long id);

    void updateSeatType(Long seatId, String seatType);

    void updateSeatStatus(Long seatId, String status);
}

File: cinemax-fe/src/features/staff/services/roomService.js
import axiosClient from '@/shared/services/axiosClient'

export function getRooms() {
  return axiosClient.get('/api/v1/rooms')
}

export function getRoomById(id) {
  return axiosClient.get(`/api/v1/rooms/${id}`)
}

export function getRoomDetail(id) {
  return axiosClient.get(`/api/v1/rooms/${id}/detail`)
}

export function createRoom(payload) {
  return axiosClient.post('/api/v1/rooms', payload)
}

export function updateRoom(id, payload) {
  return axiosClient.put(`/api/v1/rooms/${id}`, payload)
}

export function deleteRoom(id) {
  return axiosClient.delete(`/api/v1/rooms/${id}`)
}

export function updateSeatType(seatId, seatType) {
  return axiosClient.put(`/api/v1/rooms/seats/${seatId}/type`, { seatType })
}

export function updateSeatStatus(seatId, status) {
  return axiosClient.put(`/api/v1/rooms/seats/${seatId}/status`, { status })
}
